src/lsp/cobol_lsp/lsp_request.ml

```diff
@@ -454,6 +454,23 @@ let handle_semtoks_full,
 
 (** {3 Hover} *)
 
+let documetation_of_datadef ~(comments: Cobol_preproc.Text.comments) ~filename data_def =
+  let loc = Cobol_data.Item.def_loc data_def in
+  let def_range = Lsp_position.range_of_srcloc_in ~filename loc in
+  let comments = List.rev @@
+    List.filter_map begin fun ({ comment_loc; comment_kind; comment_contents }: Cobol_preproc.Text.comment) ->
+      let com_range = Lsp_position.range_of_lexloc comment_loc in
+      if def_range.start.line = com_range.start.line
+      || def_range.start.line = com_range.start.line + 1
+         && comment_kind ==  `Line
+      then Some comment_contents
+      else None
+    end comments in
+  if List.length comments == 0
```

Comment:
Please use a `match` to avoid an unnecessary traversal of `List.length` (I doubt it's removed via compiler optimization, even though it could…).

Reply:
Also I don't think the case of multiple line comments is in the tests. Also what about an item with both a line and an inline comment (I guess the latter should prevail).